packages/wallets/nufi/src/adapter.ts

```diff
@@ -0,0 +1,289 @@
+     * {@link NufiEvents.disconnect} event being emitted. */
+    isConnected: boolean;
+
+    /** Subscribe to an event. */
+    on: <T extends keyof NufiEvents>(event: T, fn: NufiEvents[T]) => void;
+
+    /** Unsubscribe from an event. */
+    off: <T extends keyof NufiEvents>(event: T, fn: NufiEvents[T]) => void;
+
+    /** Request the signing of a transaction. This might take a long time, as
+     * the user may be asked for their approval. If the promise is resolved, a
+     * new signed `Transaction` object is returned. If the promise is rejected,
+     * it can either mean that the user rejected the transaction, or any other
+     * error. It is unspecified what is returned on rejection.
+     * */
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+
+    /**
+     * The behavior is identical to that of {@link signTransaction}, except
+     * that multiple transactions are requested to be signed. The promise is
+     * only resolved if all transactions have been signed.
+     */
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+
+    /**
+     * The behavior is identical to that of {@link signTransaction}, except
+     * that if the transaction is succesfully signed, it is also submitted, and
+     * only its signature is returned.
+     */
+    signAndSendTransaction(transaction: Transaction): Promise<{ signature: TransactionSignature }>;
+
+    /**
+     * Request the signing of a message. This might take a long time, as the user
+     * may be asked for their approval. If the promise is resolved, the signature
+     * is returned. If the promise is rejected, it can either mean that the user
+     * rejected the signing the message, or any other error. It is unspecified
+     * what is returned on rejection.
+     */
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+
+    /**
+     * Request connecting to the wallet. This might take a long time, as the
+     * user may be asked for their consent. Before the promise resolves, the
+     * {@link NufiEvents.connect} event will be emitted. If the promise is
+     * rejected, it can either mean that the user rejected the connection, or
+     * any other error. It is unspecified what is returned on rejection.
+     */
+    connect(): Promise<void>;
+
+    /**
+     * Request disconnecting from the wallet. Before the promise resolves, the
+     * {@link NufiEvents.disconnect} event will be emitted. The promise is
+     * guaranteed to always resolve.
+     */
+    disconnect(): Promise<void>;
+}
+
+interface NufiWindow extends Window {
+    nufiSolana?: NufiWallet;
+}
+
+declare const window: NufiWindow;
+
+export interface NufiWalletAdapterConfig {}
+
+export const NufiWalletName = 'NuFi' as WalletName<'NuFi'>;
+
+export class NufiWalletAdapter extends BaseMessageSignerWalletAdapter {
+    name = NufiWalletName;
+    url = 'https://nu.fi';
+    icon =
```

Comment:
nit: line break here, can you run Prettier (`yarn run fmt`) on the project root?

Reply:
It is prettier that introduces this line break. (Running `yarn run fmt` does not change anything.)